src/main/java/stream/flarebot/flarebot/commands/moderation/PermissionsCommand.java

```diff
@@ -281,32 +285,60 @@ public void onCommand(User sender, GuildWrapper guild, TextChannel channel, Mess
-                        .setDescription("There are no groups for this guild!")
-                        .build()).queue();
-                return;
-            } else {
-                int page = args.length == 2 ? Integer.valueOf(args[4]) : 1;
-                Set<String> groups = this.getPermissions(channel).getGroups().keySet();
-                List<String> groupList = GeneralUtils.orderList(groups);
+        } else if (args.length >= 1) {
+            if (args[0].equalsIgnoreCase("groups")) {
+                if (this.getPermissions(channel).getListGroups().isEmpty()) {
+                    channel.sendMessage(MessageUtils.getEmbed(sender)
+                            .setColor(Color.BLUE)
+                            .setDescription("There are no groups for this guild!")
+                            .build()).queue();
+                    return;
+                } else {
+                    int page = args.length == 2 ? Integer.valueOf(args[4]) : 1;
+                    Set<String> groups = this.getPermissions(channel).getGroups().keySet();
+                    List<String> groupList = GeneralUtils.orderList(groups);
 
-                String list = getStringList(groupList, page);
-                EmbedBuilder eb = MessageUtils.getEmbed(sender);
-                eb.addField("Groups", list, false);
-                eb.addField("Current page", String.valueOf(page), true);
-                int pageSize = 20;
-                int pages =
-                        groups.size() < pageSize ? 1 : (groups.size() / pageSize) + (groups.size() % pageSize != 0 ? 1 : 0);
-                eb.addField("Pages", String.valueOf(pages), true);
-                eb.setColor(Color.CYAN);
-                channel.sendMessage(eb.build()).queue();
+                    String list = getStringList(groupList, page);
+                    EmbedBuilder eb = MessageUtils.getEmbed(sender);
+                    eb.addField("Groups", list, false);
+                    eb.addField("Current page", String.valueOf(page), true);
+                    int pageSize = 20;
+                    int pages =
+                            groups.size() < pageSize ? 1 : (groups.size() / pageSize) + (groups.size() % pageSize != 0 ? 1 : 0);
+                    eb.addField("Pages", String.valueOf(pages), true);
+                    eb.setColor(Color.CYAN);
+                    channel.sendMessage(eb.build()).queue();
+                    return;
+                }
+            } else if (args[0].equalsIgnoreCase("reset")) {
+                guild.setPermissions(new PerGuildPermissions());
+                MessageUtils.sendSuccessMessage("Successfully reset perms", channel, sender);
+                return;
+            } else if (args[0].equalsIgnoreCase("restoredefault")) {
+                if (guild.getPermissions().hasGroup("Default")) {
+                    guild.getPermissions().deleteGroup("Default");
+                }
+                guild.getPermissions().addGroup("Default");
```

Comment:
Duplicate code, make a better way to handle adding default perms. We don't want to keep changing each code instance every time we add a new default perm.

Reply:
@WalshyDev this is for is they mess up their default group and don't want to have to manually fix it. This isn't for adding new default perms. That might come in a later pr.

Reply:
I know what it does, we already do this same thing though. The place you copied it from. Instead of having dupe code make a single method and call it.

Reply:
Maybe a `resetGroup` method might work, if the passed group is default is does all that ^ if not it just empties it

Reply:
i mean i don't think we want group resetting for more then default because at that point you just re create the group

Reply:
But that is again more repeated code, better to have it all in one place

Reply:
we're not restoring other groups anywhere so it isn't repetitive code

Reply:
Fair enough, `resetDefault()` then I guess